[Lee Jia Yi] iP

jiayi1129 opened the pull request:
# QuestPro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))


QuestPro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/jiayi1129/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE**!

Features:

- [x]  Managing tasks
- [x]  Managing deadlines
- [x]  Managing events

This is how QuestPro is ran 🏃 
```Java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

jiayi1129 committed:
Level-1

jiayi1129 committed:
Level-2

jiayi1129 committed:
Level-3

jiayi1129 committed:
Level-4

jiayi1129 committed:
A-TextUiTesting

jiayi1129 committed:
Level-5

jiayi1129 committed:
Level-6

jiayi1129 committed:
A-Enums

jiayi1129 committed:
Level-7

jiayi1129 committed:
Level-8

jiayi1129 committed:
Merge branch 'branch-level-8'

jiayi1129 committed:
Merge branch 'branch-level-7'

jiayi1129 committed:
Merge branch 'branch-level-8'

jiayi1129 committed:
Merge branch 'master' of https://github.com/jiayi1129/ip

jiayi1129 committed:
Level 7: Add save function

jiayi1129 committed:
Merge branch 'branch-level-7'

jiayi1129 committed:
Fix Merge Conflicts

jiayi1129 committed:
A-MoreOOP

jiayi1129 committed:
A-Packages

jiayi1129 committed:
A-JUnit

jiayi1129 committed:
A-Jar

jiayi1129 committed:
Update JavaDocs

jiayi1129 committed:
Update imports

jiayi1129 committed:
Level-9: Add Find functionality

jiayi1129 committed:
Merge branch 'branch-A-JavaDoc'

jiayi1129 committed:
Merge branch 'branch-A-CodingStandard'

jiayi1129 committed:
Merge branch 'branch-Level-9'

Py0000 commented on src/main/java/helpers/Helpers.java:
```diff
@@ -0,0 +1,20 @@
+package helpers;
+
+public class Helpers {
```
Although it may not be necessary, but perhaps can consider to include comment/brief description of `Helpers` class?

jiayi1129 replied:
Yes I think it would be good to have a description too. Will add in future commits.

Py0000 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,89 @@
+     */
+    private static final String ERROR_UNKNOWN_COMMAND = "Unknown Command. See README for full list of commands.";
+
+    /**
+     * Returns the Command corresponding to the full String command input.
+     * String input examples:
+     * <p>
+     * bye
+     * list
+     * todo read book
+     * deadline return book /by 2021-10-15
+     * deadline return book /by 2021-10-15 18:00
+     * event book fest /at 2021-10-15
+     * event book fest /at 2021-10-15 18:00
+     * done 2
+     *
+     * @param fullCommand Full String command input by user.
+     * @return Corresponding command from the input.
+     */
```
I like how the comments are so detailed. 👍

jiayi1129 replied:
Thank you! 😄

Py0000 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,89 @@
+
+import exceptions.DukeException;
+
+public class Parser {
```
Similar to previous suggestion, although it may not be necessary, but perhaps can consider to include comment/brief description of `Parser` class?

jiayi1129 replied:
Yes I think it would be good to have a description too. Will add in future commits.

Py0000 commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * Get task string to be inputted into file.
```
Not really sure on this, but perhaps, can replace `Get` with `Gets`? Please do check if I am right.

jiayi1129 replied:
That's a good catch. Will fix it, thanks for pointing out!

Py0000 commented on src/test/java/parser/ParserTest.java:
```diff
@@ -0,0 +1,71 @@
+package parser;
+
+import command.*;
```
Perhaps can list out all the import statements explicitly?

jiayi1129 replied:
Great Point! Will push this out in the next commits!

Py0000 commented on src/test/java/task/TodoTaskTest.java:
```diff
@@ -0,0 +1,42 @@
+package task;
+
+import exceptions.DukeException;
+import org.junit.jupiter.api.Test;
```
Not sure if it is necessary, maybe you can consider to separate the import statements from different packages? Please do check if this is correct.
I have also noticed this in the other test files.

jiayi1129 replied:
Great Point! Will push this out in the next commits!

Py0000 reviewed (approved):
I like your program! Its it well coded and well commented! It is easy to read and understand the flow and logic of the codes. 👍  

Other than a few minor nitpicks, it LGTM 😄

jiayi1129 committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

jiayi1129 committed:
A-Gradle: Add Gradle Support

jiayi1129 committed:
Level 10: Add GUI

jiayi1129 committed:
Merge branch 'branch-level-10'

jiayi1129 committed:
Fix issues related to Py0000 PR review

KT27Learn commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,41 @@
+public class Duke extends Application {
+    private final String DEFAULT_FILEPATH = "data/tasks.txt";
+
+    private final Storage storage;
```
Perhaps your final variable should be in all uppercase? I've noticed this in other areas as well.
Do correct me if I'm wrong :)

jiayi1129 replied:
Good point! I'll work on this, thanks!

KT27Learn commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+    public static final String COMMAND_EVENT = "event";
+    public static final String COMMAND_DONE = "done";
+    public static final String COMMAND_DELETE = "delete";
+    public static final String COMMAND_FIND = "find";
```
Perhaps an alternative would be to declare the command strings as enum variables instead?

jiayi1129 replied:
How would you related the enum variables to the command strings then? Would it be through a Map?

KT27Learn reviewed (commented):
Everything looks great, and I think that it is an overall job well done! There are no major issues, and the code base was easy to understand. There are just some small styling suggestions that I have both brought up.

jiayi1129 committed:
Add assertions to Storage class

Empty task may be added to the TaskList while reading from Storage.

Asserting task added in load method to be not null ensures that
no null object is added to TaskList.

Empty task may be written to the storage file.

Asserting task written in writeTasksToFile method to be not empty
or null ensures that no empty strings are written to TaskList and
prevent Null Pointer errors.

jiayi1129 committed:
Change constant variables to use uppercase letters.

To maintain the project code quality and consistency
of using constants.

Changing the constant variables of lowercase letters
to all uppercase letters.

jiayi1129 committed:
Merge pull request #1 from jiayi1129/branch-A-Assertions

Add assertions to Storage class

jiayi1129 committed:
Merge branch 'master' into branch-A-Assertions

jiayi1129 committed:
Merge branch 'master' into branch-A-CodeQuality

jiayi1129 committed:
Merge pull request #2 from jiayi1129/branch-A-CodeQuality

Change constant variables to use uppercase letters.

jiayi1129 committed:
Merge branch 'master' into branch-A-Assertions

jiayi1129 committed:
Merge branch 'master' into branch-A-CodeQuality

jiayi1129 committed:
Add C-Help functionality

jiayi1129 committed:
Merge branch 'branch-BCD-Extension'

jiayi1129 committed:
Merge branch 'branch-A-Assertions'

jiayi1129 committed:
Merge branch 'branch-A-CodeQuality'

jiayi1129 committed:
A-BetterGui: Improve GUI

jiayi1129 committed:
Merge branch 'branch-A-BetterGui'

jiayi1129 committed:
Change product name

jiayi1129 committed:
Add product screenshot

jiayi1129 committed:
Update README and gradle

jiayi1129 committed:
Merge branch 'branch-A-UserGuide'

jiayi1129 committed:
Update README

jiayi1129 committed:
Update README
